### PR TITLE
Add preliminary command line options to main.cpp

### PR DESCRIPTION
**Pull request by cmhobbs**
- Add -h, --help, -v, and --version
- Address most of Issue #219.
- Update debug docs for consistency.

### CODE REVIEW

**Comment by cmhobbs** [comment]
Please note that I did not implement the `--data-dir` option because I do not yet know enough C++ to track down the paths.  This should at least resolve part of the issue.  Maybe a separate issue for the `--data-dir` switch could be created?

File: SourceX/main.cpp [hunk 1]
```diff
@@ -19,13 +19,53 @@ static std::string build_cmdline(int argc, char **argv)
...
+		return 0;
+	} else if ((cmdline == "-v") || (cmdline == "--version")) {
+	  // TODO: replace this with a more robust variable rather
+	  //       than hard coding it here.  this is also defined in
+	  //       Packaging/fedora/devilutionx.spec and
+	  //       docs/CHANGELOG.md.  VerQeuryValueA was suggested in
+	  //       GitHub issue #219.
+	  fprintf(stdout, "devilutionX v0.4.0.\n");
+	  fprintf(stdout, "Diablo build for modern operating systems.\n");
+	  fprintf(stdout, "https://github.com/diasurgical/devilutionX/\n");
```
Comment (Manuel-K) [inline SourceX/main.cpp]: This branch misses a return but it is not a `void` function.

Comment (Manuel-K) [reply SourceX/main.cpp]: but it is not a `void` function.

File: SourceX/main.cpp [hunk 2]
```diff
@@ -19,13 +19,53 @@ static std::string build_cmdline(int argc, char **argv)
...
+	} else {
+		return dvl::WinMain(NULL, NULL, (char *)cmdline.c_str(), 0);
```
Comment (Manuel-K) [inline SourceX/main.cpp]: If both if clauses contain a return, you don't need the else – you can leave that part unchanged.

Comment (cmhobbs) [reply SourceX/main.cpp]: Pardon my ignorance here, but again, C++ is pretty new to me.

What's the best way to handle this?  Remove the `else` statement and add a return to the `else if`?  What type of return value other than `0` would be appropriate given this is an `int` function?

Thanks for looking at this!

Comment (qndel) [reply SourceX/main.cpp]: he probably meant that you could skip skip the 
`else {return dvl::WinMain(NULL, NULL, (char *)cmdline.c_str(), 0);}`
and do just `return dvl::WinMain(NULL, NULL, (char *)cmdline.c_str(), 0);`
but it's fine imo

Comment (cmhobbs) [reply SourceX/main.cpp]: How about now?

File: SourceX/main.cpp (hunk 1 above)
Comment (AJenbo) [reply SourceX/main.cpp]: Add `return 0;` here so that the game doesn't start when checking the verison.

File: SourceX/main.cpp [hunk 3]
```diff
@@ -19,13 +19,53 @@ static std::string build_cmdline(int argc, char **argv)
...
+		fprintf(stdout, "    %-20s %-30s\n", "-m <##>", "add debug monster, up to 10 allowed");
+		fprintf(stdout, "    %-20s %-30s\n", "-q <#>", "force a certain quest");
+		fprintf(stdout, "    %-20s %-30s\n", "-r <##########>", "set map seed to");
+		fprintf(stdout, "    %-20s %-30s\n", "-t <##>", "set current quest level");
+		fprintf(stdout, "\nMultiplayer settings:\n");
+		fprintf(stdout, "    %-20s %-30s\n", "TCP/IP", "host is required to expose port 6112");
+		fprintf(stdout, "    %-20s %-30s\n", "UDP/IP", "all players are required to expose port 6112");
+		fprintf(stdout, "\nhttps://github.com/diasurgical/devilutionX/\n");
+		return 0;
+	} else if ((cmdline == "-v") || (cmdline == "--version")) {
```
Comment (AJenbo) [inline SourceX/main.cpp]: Remove the else since the previous condition has a return.

**Comment by AJenbo** [comment]
Merged with edits here https://github.com/diasurgical/devilutionX/commit/436e25207aa5961e0a89d47104fdfed710739ecb

**Comment by AJenbo** [comment]
Also merged upstream: https://github.com/diasurgical/devilution/commit/ae52d223d1356db508cb257415754b22dc61521f

**Pull request closed without merging**